.ci-support/merge-install-production-branch.sh

```diff
@@ -12,28 +12,65 @@ set -x
 #
 # PRODUCTION_BRANCH = The production branch (default=production)
 # PRODUCTION_FORK = The production fork (default=illinois-ceesd)
-#
-MIRGE_HOME=${1:-"."}
-PRODUCTION_BRANCH=${PRODUCTION_BRANCH:-"production"}
-PRODUCTION_FORK=${PRODUCTION_FORK:-"illinois-ceesd"}
 
+MIRGE_HOME=${1:-"."}
 echo "MIRGE_HOME=${MIRGE_HOME}"
+
+if [[ "$GITHUB_HEAD_REF" == "$PR_MAIN_BRANCH" ]]; then
+  # GITHUB_HEAD_REF is only set for PR builds.
+  # https://docs.github.com/en/actions/learn-github-actions/environment-variables
+  PRODUCTION_BRANCH="production-merge/${PR_MAIN_BRANCH}"
+  PRODUCTION_FORK=${PR_PRODUCTION_FORK:-"illinois-ceesd"}
+else
+  PRODUCTION_BRANCH="${PRODUCTION_BRANCH:-"production"}"
+  PRODUCTION_FORK=${PRODUCTION_FORK:-"illinois-ceesd"}
+fi
```

Comment:
This seems to indicate PRs *must* have a compatible production branch instead of allowing a default to regular ole production.   If it merges, it seems it will break all production feeder branches - none of which require such a custom production branch.

Reply:
That's not the intention. What makes you think that? To me, the logic is that "if we're currently building a PR *and* if the production env settings indicate that a special branch is needed (by setting `PR_MAIN_BRANCH` to the name of the current branch... come to think of it, I'll probably rename that variable, likely `PRODUCTION_PR_MAIN_BRANCH`), then use that branch. Otherwise, fall back to the old logic.

Reply:
The previous logic was if the PR requires a special production branch, then they set PRODUCTION_BRANCH.  I am unsure why additional variables are needed.

Reply:
No, the original variables stay unchanged. The additional variable (now `PRODUCTION_PR_MAIN_BRANCH`) contains the name of the branch headed into `main`. That way, the CI script can tell whether it's currently testing that branch and, if so, test the corresponding production branch (found from `PRODUCTION_PR_MAIN_BRANCH` by convention) by appropriately populating the existing variables. Likewise, `PRODUCTION_PR_FORK` is needed because it's only supposed to be active *if* we're still on the specific branch.

Reply:
I do trust you know what you are doing; I just don't understand.   

When CI runs these scripts, it is always testing the current branch, so I didn't find it necessary to indicate to CI what branch it is currently testing.   The production testing procedure was to merge "production" into the current branch (i.e. the branch being tested, regardless of what branch it is)  and run the production cases, and any branch could specify a custom branch to use for "production" through the PRODUCTION_BRANCH variable.

It appeared that this change set removes the PRODUCTION_BRANCH variable hooks from `production_testing_env.sh`, and force the setting of PRODUCTION_PR_MAIN_BRANCH.   The previous logic would use `production` proper iff the PRODUCTION_BRANCH var was not set.